### DISCUSSION TITLE
How to store the 'intended RF use' parameter

### DISCUSSION THREAD

**Discussion by m-a-x-i-m-z** [pulseq/pulseq#70] [Ideas]
Dear all, 

I just was to quickly get forum's opinion to the question: how should we store the RF use flag of the RF pulse? Currently Matlab Pulseq supports the following RF use options: 

`supported_rf_use={'excitation','refocusing','inversion','saturation','preparation'};` 

With regard to the storage format, the options are: 

1. a single number in the RF pulse description, relying on a predefined mapping, e.g. 0:undefined, 1:excitation, etc. 
2. a single number in the RF pulse description, referring to the table of strings, for which we need to define a new object/section
3. a single character in the RF pulse description (e.g. first character of the current supported types) 
4. a full (or abbreviated) string in the in the RF pulse description.

The latter option is most human-readable, but will substantially increase the file volume. Option 3 is almost as readable. Options 3 and 4 will change in principle the storage format of the RF pulse, because currently we only have numbers in the RF section (but for the Option 3 the change is small). Options 4 and 2 are most flexible and extensible in the future, e.g. if we decide to add new RF use types. Option 2 is as compact as Option 3 in terms of the memory footprint. Option 1 is the easiest to implement, but will be very illegible and hard to extend. 

In terms of the pulse uses, do we need something additional right away? I would extend the list above with two options right away: 'undefined' and 'other'. And as for these 7 use types the first letter still uniquely identifies the respective words, my personal favorite is the Option 3. 

Best, 
Maxim

**Comment by fsantini** [comment]
Dear Maxim,

Option 3 would be my preferred one too.

Cheers,

Francesco

**Comment by QingpingChen1201** [comment]
Hi Maxim,
to use a single number/character to refer to a specific string in the table of all supported RF types sounds good to me:-)
Best regards,
Qingping

**Reply by m-a-x-i-m-z** [reply]
Option 3 in my proposal consists in having one character in the RF pulse description and no additional table stored in the file. But we can add a comment to make the whole thing really human readable

**Comment by zhengliuX** [comment]
Hi Maxim,
    Option 1 & 3 sounds good to me, too. Option 1 is more like a machine language, so I guess Option 3 is more friendly. As for the info is stored in the file, does this mean we could get the info in the C++ interface? Like `rfEvent.GetUsage()`? 

Best,
Zheng

**Reply by m-a-x-i-m-z** [reply]
Yes, it will eventually be added to the C++ interface, although we currently have no pressing use case for it. 
--M.
